src/main/java/graphql/execution/nextgen/depgraph/DependencyGraph.java

```diff
@@ -0,0 +1,123 @@
+package graphql.execution.nextgen.depgraph;
+
+import graphql.language.OperationDefinition;
+import graphql.schema.GraphQLSchema;
+import graphql.util.TraversalControl;
+import graphql.util.Traverser;
+import graphql.util.TraverserContext;
+import graphql.util.TraverserVisitorStub;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+public class DependencyGraph {
+
+
+    public DependencyGraph createDependencyGraph(GraphQLSchema graphQLSchema, OperationDefinition operationDefinition) {
+
+        FieldCollectorWTC fieldCollector = new FieldCollectorWTC();
+        FieldCollectorParameters parameters = FieldCollectorParameters
+                .newParameters()
+                .schema(graphQLSchema)
+                .build();
+
+        Function<MergedFieldWithType, List<MergedFieldWithType>> getChildren = mergedFieldWithType -> {
+            List<MergedFieldWithType> childs = fieldCollector.collectFields(parameters, mergedFieldWithType);
+            return childs;
+        };
+
+
+        Set<FieldVertex> allVertices = new LinkedHashSet<>();
+
+        Traverser<MergedFieldWithType> traverser = Traverser.depthFirst(getChildren);
+        FieldVertex rootVertex = new FieldVertex(null, null, null, null, null);
+        traverser.rootVar(FieldVertex.class, rootVertex);
+        allVertices.add(rootVertex);
+        List<MergedFieldWithType> roots = fieldCollector.collectFromOperation(parameters, operationDefinition, graphQLSchema.getQueryType());
+        traverser.traverse(roots, new TraverserVisitorStub<MergedFieldWithType>() {
```

Comment:
I think recursive collecting fields (graph traversal!) and then traversing collected fields again to build a graph is quite redundant. I would suggest to build the graph in 1 traverse cycle, incrementally with field collection.